src/proposal.mligo

```diff
@@ -134,10 +148,25 @@ let stake_tk(token_amount, addr, store : nat * address * storage): storage =
   in { store with freeze_history = new_freeze_history }
 
 [@inline]
-let unfreeze_on_ledger (tokens, addr, ledger, total_supply, unfrozen_token_id, frozen_token_id : nat * address * ledger * total_supply * token_id * token_id): (ledger * total_supply) =
+let unfreeze_on_ledger (tokens, addr, ledger, total_supply, frozen_token_id, token_address : nat * address * ledger * total_supply * token_id * address): (operation * ledger * total_supply) =
+  // Call transfer on token_contract to transfer `token` number of
+  // tokens from `addr` to the address of this contract. Right now token_id `0`
+  // is used, but we should probably confirm this TODO@SRAS
```

Comment:
Indeed we should not assume that `0` is always going to be the correct `token_id`, we should instead have this in the storage and known at origination alongside `token_address`, I'm thinking that we should replace `token_address` with an `governance_token` that itself contains an `address` and `token_id` fields.

Reply:
Sounds good. Have replaced `token_address` with `governance_token` field of type `{address: address, token_id: nat}`, Have changed code, makefile and docs accordingly.